include/fiction/utils/layout_utils.hpp

```diff
@@ -350,7 +352,54 @@ Lyt convert_to_fiction_coordinates(const sidb_cell_clk_lyt_siqad& lyt) noexcept
 
     return lyt_new;
 }
+/**
+ * Generates a random coordinate within the region spanned by two given coordinates. The two given coordinates form the
+ * lower left corner and the upper right corner of the spanned region.
+ *
+ * @OffsetCoordinateType The coordinate implementation to be used.
+ * @param coordinate1 Coordinate of the lower left corner.
+ * @param coordinate2 Coordinate of the upper right corner (coordinate order is not important, automatically swapped if
+ * necessary).
+ * return Randomly generated coordinate.
+ */
+template <typename CoordinateType>
+CoordinateType random_coordinate(const CoordinateType& coordinate1, const CoordinateType& coordinate2) noexcept
+{
+    static std::mt19937_64 generator(std::random_device{}());
+
+    CoordinateType left_coordinate  = coordinate1;
+    CoordinateType right_coordinate = coordinate2;
```

Comment:
As mentioned in the last review, if you pass the parameters by copy, you don't have to allocate extra variables.

Reply:
Sorry! Now, I got it.

Reply:
I tried it but then, swap does not work anymore.

Reply:
It will if you pass the parameters by copy and remove `const`.

Reply:
True! Thanks!